tests/foreman/cli/test_computeresource_rhev.py

```diff
@@ -17,30 +17,94 @@
 from robottelo.cli.factory import CLIFactoryError
 from robottelo.cli.factory import CLIReturnCodeError
 from robottelo.cli.factory import make_compute_resource
+from robottelo.cli.factory import make_host
 from robottelo.cli.factory import make_os
+from robottelo.cli.host import Host
+from robottelo.cli.settings import Settings
 from robottelo.config import settings
+from robottelo.constants import RHEL_6_MAJOR_VERSION
+from robottelo.constants import RHEL_7_MAJOR_VERSION
+from robottelo.helpers import host_provisioning_check
+from robottelo.utils.issue_handlers import is_open
 
 pytestmark = pytest.mark.on_premises_provisioning
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="module")
 def rhev():
-    rhev = type('rhev', (object,), {})()
-    rhev.current_rhev_url = settings.rhev.hostname
-    rhev.username = settings.rhev.username
+    bridge = settings.vlan_networking.bridge
+    rhev = type("", (), {})()
+    rhev.url = settings.rhev.hostname
     rhev.password = settings.rhev.password
+    rhev.username = settings.rhev.username
     rhev.datacenter = settings.rhev.datacenter
+    rhev.image_name = settings.rhev.image_name
     rhev.image_arch = settings.rhev.image_arch
     rhev.image_uuid = settings.rhev.image_uuid
-    rhev.os = make_os()
+    rhev.image_os = settings.rhev.image_os
+    rhev.image_user = settings.rhev.image_username
+    rhev.image_pass = settings.rhev.image_password
+    rhev.vm_name = settings.rhev.vm_name
+    rhev.storage_domain = settings.rhev.storage_domain
+    rhev.rhv_api = RHEVMSystem(
+        hostname=rhev.url.split('/')[2],
+        username=rhev.username,
+        password=rhev.password,
+        version='4.0',
+        verify=False,
+    )
+    rhev.cluster_id = rhev.rhv_api.get_cluster(rhev.datacenter).id
+    rhev.storage_id = rhev.rhv_api.get_storage_domain(rhev.storage_domain).id
+    if bridge:
+        rhev.network_id = (
+            rhev.rhv_api.api.system_service()
+            .networks_service()
+            .list(search=f'name={bridge}')[0]
+            .id
+        )
+    if is_open('BZ:1685949'):
+        dc = rhev.rhv_api._data_centers_service.list(search=f'name={rhev.datacenter}')[0]
+        dc = rhev.rhv_api._data_centers_service.data_center_service(dc.id)
+        rhev.quota = dc.quotas_service().list()[0].id
+    else:
+        rhev.quota = 'Default'
     return rhev
 
 
+@pytest.fixture(scope='module')
+def os():
+    return make_os()
+
+
+@pytest.fixture(scope="module")
+def provisioning(module_org, module_location):
+    provisioning = type("", (), {})()
+    provisioning.org_name = module_org.name
+    provisioning.loc_name = module_location.name
+    provisioning.config_env = configure_provisioning(
+        compute=True, org=module_org, loc=module_location, os=None
+    )
+    provisioning.os_name = provisioning.config_env['os']
+    return provisioning
+
+
+@pytest.fixture(scope="module")
+def tear_down(provisioning):
```

Comment:
I know this might be painful, but we should make sure the VM got removed from the CR and remove it using its API otherwise.
(there's a potential scenario, where foreman doesn't clean the VM up)

Not a blocker, perhaps, just some food for thought.